benchmark/polyfills.js

```diff
@@ -0,0 +1,187 @@
+'use strict';
+
+/**
+ * Polyfills for older Node.js versions (0.8+)
+ * This file provides implementations for features not available in older Node versions
+ */
+
+// Polyfill for Array.from (ES6)
+if (!Array.from) {
+    Array.from = function (arrayLike, mapFn, thisArg) {
+        var result = [];
+        var length, i;
+
+        if (arrayLike == null) {
+            throw new TypeError('Array.from requires an array-like object');
+        }
+
+        length = arrayLike.length >>> 0;
+
+        for (i = 0; i < length; i++) {
+            if (i in arrayLike) {
+                if (mapFn) {
+                    result[i] = mapFn.call(thisArg, arrayLike[i], i, arrayLike);
+                } else {
+                    result[i] = arrayLike[i];
+                }
+            }
+        }
+
+        return result;
+    };
+}
+
+// Polyfill for String.prototype.repeat (ES6)
+if (!String.prototype.repeat) {
+    String.prototype.repeat = function (count) {
+        if (count < 0) {
+            throw new RangeError('repeat count must be non-negative');
+        }
+        if (count === Infinity) {
+            throw new RangeError('repeat count must be less than infinity');
+        }
+
+        count = Math.floor(count);
+        if (this.length === 0 || count === 0) {
+            return '';
+        }
+
+        var result = '';
+        var pattern = this;
+
+        while (count > 0) {
+            if (count & 1) {
+                result += pattern;
+            }
+            count >>= 1;
+            pattern += pattern;
+        }
+
+        return result;
+    };
+}
+
+// Polyfill for Object.fromEntries (ES2019)
+if (!Object.fromEntries) {
+    Object.fromEntries = function (entries) {
+        var obj = {};
+
+        for (var i = 0; i < entries.length; i++) {
+            var entry = entries[i];
+            if (Object.prototype.toString.call(entry) !== '[object Array]') {
+                throw new TypeError('Iterator value ' + entry + ' is not an entry object');
+            }
+            obj[entry[0]] = entry[1];
+        }
+
+        return obj;
+    };
+}
+
+// Polyfill for Object.entries (ES2017)
+if (!Object.entries) {
+    Object.entries = function (obj) {
+        var ownProps = Object.keys(obj);
+        var i = ownProps.length;
+        var resArray = new Array(i);
```

Comment:
as a side note, never use `new Array`, ever, as it makes a permanently slow object.